src/libcollections/vec_deque.rs

```diff
@@ -1396,6 +1396,38 @@ impl<T> VecDeque<T> {
         // naive impl
         self.extend(other.drain());
     }
+
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all elements `e` such that `f(&e)` returns false.
+    /// This method operates in place and preserves the order of the retained
+    /// elements.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut buf = VecDeque::new();
+    /// buf.extend(1..4);
+    /// buf.retain(|&x| x%2 == 0);
+    /// assert_eq!(buf, [2, 4]);
+    /// ```
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
At this point anything to be included in 1.0 needs to be cherry-picked backwards, and this isn't necessarily a serious enough change to be cherry-picked backwards, so if we're going to insta-stabilize it this should be `since = "1.1.0"` with a unique feature name like `vec_deque_retain`.